src/Mentat.UI/Controllers/CardController.cs

```diff
@@ -75,9 +98,17 @@ public ActionResult SaveCard(string id, [FromForm] Card card, [FromForm] string
         }
 
         // GET: CardController/Delete/5
-        public ActionResult Delete(string id)
+        public ActionResult Delete(string id, string user)
         {
             var card = _cardService.GetCard(id);
+            if (card.Owner != user)
+            {
+                return NotFound($"{user} is not the owner of this card and cannot delete it.");
+            }
+            if (!card.IsPrivate)
```

Comment:
This check doesn't make much sense to me. Why am I restricted from deleting my own card regardless of its privacy?